packages/checkout/widgets-lib/src/lib/hooks/useDebounce.ts

```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from 'react';
+
+const useDebounce = (value, delay = 500) => {
```

Comment:
Any reason you've gone with making this a hook with useEffect rather than just a function like debounce(fn, delay)?

Reply:
My thinking is that a custom hook encapsulates the functionality well and cleans up after itself hooking into Reacts lifecycle, the other benefit is hooking into Reacts state/effects will then easily react to changes in `value` and `delay` without having to manage any of that in every instance it's used.. flushing and syncing the debounce.

Reply:
Just been thinking of ways we can leverage React more with things like custom hooks to reduce boilerplate and shared functionality across our Widgets.. this felt like a nice baby step towards that :)

Reply:
My thinking was around the fact that we will be calling the function which sets the value to be used in debounce over and over and only the debouncedValue will update after the delay, versus, debouncing the actual function which will only be called once (after the delay) to set the value.

With how we are using this in the swap widget, we will call setAmount() on every input keystroke and then only the debouncedValue updates after the delay and I was expecting that we would debounce the inputChange so that setAmount() is only called once after the delay.

The difference will be in the amount of re-renders happening as we are still setting the 'amount' piece of state over and over.

Reply:
It's a good callout.. from my experience delaying the data change has other impacts to the UX that can be harder to map to, but the fetch is really the key of what we need to debounce.

Reply:
just from testing out locally seems like the re-rendering isnt too bad